src/en/cmanhua/src/eu/kanade/tachiyomi/extension/en/cmanhua/CManhua.kt

```diff
@@ -0,0 +1,612 @@
+package eu.kanade.tachiyomi.extension.en.cmanhua
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.tryParse
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.contentOrNull
+import kotlinx.serialization.json.jsonPrimitive
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.TimeZone
+
+class CManhua : HttpSource() {
+    override val name = "CManhua"
+    override val baseUrl = "https://cmanhua.com"
+    override val lang = "en"
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient.newBuilder()
+        .addInterceptor { chain ->
+            val request = chain.request()
+            if (request.header("Referer") == null) {
+                val newRequest = request.newBuilder()
+                    .header("Referer", baseUrl)
+                    .build()
+                return@addInterceptor chain.proceed(newRequest)
+            }
+            chain.proceed(request)
+        }
+        .build()
+    private val requestHeaders = headersBuilder().add("Referer", baseUrl).build()
+
+    override fun popularMangaRequest(page: Int): Request = listRequest(page, sort = SORT_TOP_VIEWS)
+
+    override fun latestUpdatesRequest(page: Int): Request = listRequest(page, sort = SORT_UPDATE_TIME)
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotBlank()) {
+            return searchRequest(page, query.trim())
+        }
+
+        var status = STATUS_OPTIONS.first().second
+        var sort = SORT_OPTIONS.first().second
+        var minChapters = CHAPTER_OPTIONS.first().second
+        var gender = GENDER_OPTIONS.first().second
+        var genres = ""
+
+        filters.forEach { filter ->
+            when (filter) {
+                is StatusFilter -> status = filter.toUriPart()
+                is SortFilter -> sort = filter.toUriPart()
+                is ChapterCountFilter -> minChapters = filter.toUriPart()
+                is GenderFilter -> gender = filter.toUriPart()
+                is GenreFilter -> {
+                    genres = filter.state.filter { it.state }.joinToString(",") { it.id }
```

Comment:
The default is `", "` and the extra space doesn't matter in this context.

```suggestion
                    genres = filter.state.filter { it.state }.joinToString() { it.id }
```